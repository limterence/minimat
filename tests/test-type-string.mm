/* test-type-string: string literals, relationals and functions */
string gets(string prompt) {     /* string function definition and argument */
  printstring(prompt);
  return("goodbye");
}
int main() {
  string s;
  string t;
  s = "hello, world\n";          /* literals */
  printstring(s);
  t = s;                         /* assignment */
  s = gets("Enter the word \"goodbye\" without quotes: ");
  printstring(s);
  printstring(t);
  printbool(s == "goodbye");     /* literals */
  printbool(s != "goodbye");
  printint(int_of_string("0500"));      /* conversions */
  printfloat(float_of_string("1.0"));
  println();
}

