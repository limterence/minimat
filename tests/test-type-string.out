hello, world
 Enter the word "goodbye" without quotes:  goodbye hello, world
 1 0 500   1.00 
